src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotHaveIncorrectSuffix.cs

```diff
@@ -56,13 +62,219 @@ public abstract class IdentifiersShouldNotHaveIncorrectSuffixAnalyzer : Diagnost
                                                                              DiagnosticSeverity.Warning,
                                                                              isEnabledByDefault: false,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: HelpLinkUri,
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(TypeNoAlternateRule, MemberNewerVersionRule, TypeNewerVersionRule, MemberWithAlternateRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            TypeNoAlternateRule,
+            MemberNewerVersionRule,
+            TypeNewerVersionRule,
+            MemberWithAlternateRule);
+
+        internal const string AttributeSuffix = "Attribute";
+        internal const string CollectionSuffix = "Collection";
+        internal const string DictionarySuffix = "Dictionary";
+        internal const string EventArgsSuffix = "EventArgs";
+        internal const string EventHandlerSuffix = "EventHandler";
+        internal const string ExSuffix = "Ex";
+        internal const string ExceptionSuffix = "Exception";
+        internal const string NewSuffix = "New";
+        internal const string PermissionSuffix = "Permission";
+        internal const string StreamSuffix = "Stream";
+        internal const string DelegateSuffix = "Delegate";
+        internal const string EnumSuffix = "Enum";
+        internal const string ImplSuffix = "Impl";
+        internal const string CoreSuffix = "Core";
+        internal const string QueueSuffix = "Queue";
+        internal const string StackSuffix = "Stack";
+
+        // Dictionary that maps from a type name suffix to the set of base types from which
+        // a type with that suffix is permitted to derive.
+        private static readonly ImmutableDictionary<string, ImmutableArray<string>> s_suffixToBaseTypeNamesDictionary = ImmutableDictionary.CreateRange(
```

Comment:
Are these dictionaries not shareable with the rule that @basoundr wrote IdentifiersShouldHaveCorrectSuffix?